### PR TITLE
Toggle Fields (CheckBoxField, RadioButtonField, etc.)

### PR DESCRIPTION
**Pull request by corranwebster**
This PR adds additional fields with toggling behaviour: check boxes, radio buttons, and toggle buttons.

The purpose of this functionality is to abstract the common API to permit an eventual refactor of the TraitsUI `BooleanEditor` to allow styles other than checkboxes.

### CODE REVIEW

File: docs/source/fields.rst [hunk 1]
```diff
@@ -62,4 +62,25 @@ values, a :py:attr:`~pyface.fields.i_spin_field.IComboField.formatter` function
 should be provided - this defaults to either :py:func:`str` (Python 3+) or
 :py:func:`unicode` (Python 2).
 
+ToggleField and Subclasses
+==========================
 
+The :py:class:`~pyface.fields.i_toggle_field.IToggleField` interface holds a
+boolean :py:attr:`~pyface.fields.i_spin_field.IToggleField.value` that is
```
Comment (rahulporuri) [inline docs/source/fields.rst:69]: should this be 

```suggestion
boolean :py:attr:`~pyface.fields.i_toggle_field.IToggleField.value` that is
```

File: pyface/fields/toggle_field.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+#
+# Author: Enthought, Inc.
+# Description: <Enthought pyface package component>
```
Comment (rahulporuri) [inline pyface/fields/toggle_field.py:12]: nit: do we really need these or were they copy-pasted from some other module?

**Review by rahulporuri — APPROVED** [review]
Mostly LGTM with a couple of small comments

File: pyface/fields/toggle_field.py (hunk 2 above)
Comment (corranwebster) [reply pyface/fields/toggle_field.py:12]: Copy-pasted.  I don't think we've done the header normalization for pyface: some of the copyright dates were also out of sync.

**Pull request merged**